pkg/component/controller/extensions_controller.go

```diff
@@ -223,13 +226,36 @@ func (cr *ChartReconciler) Reconcile(ctx context.Context, req reconcile.Request)
 	cr.L.Debugf("Installed or updated reconciliation request: %s", req)
 	return reconcile.Result{}, nil
 }
+
 func (cr *ChartReconciler) uninstall(ctx context.Context, chart v1beta1.Chart) error {
 	if err := cr.helm.UninstallRelease(ctx, chart.Status.ReleaseName, chart.Status.Namespace); err != nil {
 		return fmt.Errorf("can't uninstall release `%s/%s`: %w", chart.Status.Namespace, chart.Status.ReleaseName, err)
 	}
 	return nil
 }
 
+func removeFinalizer(ctx context.Context, c client.Client, chart *v1beta1.Chart) error {
+	idx := slices.Index(chart.Finalizers, finalizerName)
+	if idx < 0 {
+		return nil
+	}
+
+	path := fmt.Sprintf("/metadata/finalizers/%d", idx)
+	patch, err := json.Marshal([]struct {
+		Op    string `json:"op"`
+		Path  string `json:"path"`
+		Value string `json:"value,omitempty"`
+	}{
+		{"test", path, finalizerName},
+		{"remove", path, ""},
+	})
+	if err != nil {
+		return err
+	}
+
+	return c.Patch(ctx, chart, client.RawPatch(types.JSONPatchType, []byte(patch)))
```

Comment:
Isn't patch `[]byte` already?

Reply:
Wondering that there's no lint for such things.